File: app/src/androidTest/java/com/bopr/android/smailer/BaseTest.java
package com.bopr.android.smailer;

import android.content.Context;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.platform.app.InstrumentationRegistry;

import org.junit.Before;
import org.junit.runner.RunWith;

import java.util.Locale;
import java.util.TimeZone;

/**
 * Base tester.
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
@RunWith(AndroidJUnit4.class)
public abstract class BaseTest {

    protected static final String TAG = "TEST";

    protected BaseTest() {
        Locale.setDefault(Locale.US);
        TimeZone.setDefault(TimeZone.getTimeZone("America/New_York"));
    }

    protected static Context getContext() {
        return InstrumentationRegistry.getInstrumentation().getTargetContext();
    }

    @Before
    public void setUp() throws Exception {
        System.setProperty("dexmaker.dexcache", getContext().getCacheDir().getPath());
    }
}

File: app/src/androidTest/kotlin/com/bopr/android/smailer/BaseTest.kt
package com.bopr.android.smailer

import androidx.test.platform.app.InstrumentationRegistry
import androidx.work.WorkManager
import com.bopr.android.smailer.data.Database.Companion.DATABASE_NAME
import org.junit.BeforeClass
import java.util.Locale
import java.util.TimeZone


/**
 * Base tester.
 *
 * @author Boris Pronin ([boris280471@gmail.com](mailto:boris280471@gmail.com))
 */
abstract class BaseTest {

    companion object {

        val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
        fun getString(resId: Int) = targetContext.getString(resId)
        fun getString(resId: Int, vararg args: Any?) = targetContext.getString(resId, args)

        @BeforeClass
        @JvmStatic
        fun setUpClass() {
            /* cancel all running jobs to prevent interference */
            WorkManager.getInstance(targetContext).cancelAllWork()

            Locale.setDefault(Locale.US)
            TimeZone.setDefault(TimeZone.getTimeZone("America/New_York"))

            DATABASE_NAME = "test.sqlite"
        }
    }
}